### PR TITLE
[VENDOR] Update Dependencies LRN-32078

### PR DESCRIPTION
**Pull request by rhiannon-eldridge-lrn**
## Checklist

- [ ] Feature
- [ ] Bug
- [X] Security
- [X] Documentation

- [x] ChangeLog.md updated

- [ ] Tests added
- [x] All testsuites passed
- [ ] `make dist` completed successfully

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 language: python
 python:
-  - "2.7"
-  - "3.4"
   - "3.5"
```
Comment (david-scarratt-lrn) [inline .travis.yml:3]: 3.5 can be dropped.

File: README.md [hunk 2]
```diff
@@ -18,11 +18,12 @@ Supports:
 
 These are the versions we test for:
```
Comment (david-scarratt-lrn) [inline README.md:19]: See comment in ticket regarding linking to supported versions, rather than listing.

File: tox.ini [hunk 3]
```diff
@@ -4,7 +4,7 @@
 # and then run "tox" from this directory.
 
 [tox]
-envlist = py27, py36, py37
+envlist = py35, py36, py37, py38, py39
```
Comment (david-scarratt-lrn) [inline tox.ini:7]: `py35` can be removed.

**Review by david-scarratt-lrn — APPROVED** [review]

**Comment by david-scarratt-lrn** [comment]
The Makefile changes may seem over-engineered. The point is to provide an environment that facilitates development and testing in whatever configuration a developer prefers: someone with Python experience and _n_ Python versions installed locally should be able to work with a standard Python virtual environment, while someone doing FR with little Python experience and no local (standard) installation should be able to spin up a standard environment and run tests there. This complexity is pushed into the Makefile, so the developer shouldn't need to worry (too much) about it. The structure is explained in the notes at the end of the Makefile, but basically it should be usable the same as other SDKs.

**Comment by shtrom** [comment]
I think we should mandate one development configuration. Being too flexible is nice to the developer, but this means we might have to support a large combination of different issues that are linked to any custom build setup rather than the code.

Could we instead try to unify, across the SDKs, to something like a simple `docker -v` from a official upstream container for the language?

**Comment by david-scarratt-lrn** [comment]
That is how the Java and Ruby SDKs are set up (actually, Ruby is still in code review). It would have been the same here except that `test-integration-dev` just runs tox, and `tox.ini` contains all the supported Python versions. Running in a single-version container will result in errors from tox. I am less concerned about the support aspect than about having to keep `tox.ini` and the the multi-version container up to date every time the versions change.

Maybe it would be better to remove the envlist from `tox.ini` and run `tox -e<version>`, where the developer specifies the version (or a default is used). Then the container and tox would match. It loses the nice multi-env aspect of tox, but it's not hard to run `make test` in a loop over versions instead. Then only the default version would need to be updated when versions change, like the other SDKs.

**Comment by david-scarratt-lrn** [comment]
We decided that this PR, which is supposed to be for a security patch, was being taken over by a separate issue. I have taken my changes out and put them in a separate PR, and reverted this one. Apologies for the mess.

**Review by aaron-ji-lrn — COMMENTED** [review]
looks good, will do func

**Review by aaron-ji-lrn — APPROVED** [review]
It's good to go

**Pull request merged**